### PR TITLE
Removing _open_shelve 

### LINKED ISSUE #2040: utils.data._open_shelve not needed in Python 3.4
`shelve` context manager is fixed in Python 3.4:
http://bugs.python.org/issue13896

### PR DESCRIPTION
**Pull request by bsipocz**
This is a PR to close #2040. Can only be merged once 2.0 is out as the workaround is needed for py <3.4

### CODE REVIEW

**Comment by pllim** [comment]
Failure seems real:
```
NameError: name 'shelve' is not defined
```

**Comment by bsipocz** [comment]
@pllim - I should have skipped CI, this is expected to fail on everything earlier than py3.4. I assume we revise the travis matrix once the 2.0 is out and remove both 2.7 and 3.3 tests.

**Comment by pllim** [comment]
Ah, yes. Of course. Sorry for the noise!

**Comment by saimn** [comment]
The `shelve` import was local in `_open_shelve` so it's 	plausible that it is no more defined with your change (it is not in the global imports).

**Comment by bsipocz** [comment]
Oh, of course, I meant to move the import to the top.

**Comment by bsipocz** [comment]
@pllim - rebased, but conflict was genuine, so we may need to wait for travis.

File: astropy/utils/data.py [hunk 1]
```diff
@@ -1343,26 +1344,6 @@ def _get_download_cache_locs():
...
-
-    shelf = shelve.open(shelffn, protocol=2)
-
-    if withclosing:
-        return contextlib.closing(shelf)
-    else:
-        return shelf
-
-
-# the cache directory must be locked before any writes are performed.  Same for
```
Comment (pllim) [inline astropy/utils/data.py]: Just not sure if this comment line should be removed, as it is not attached to the function that was removed but the one after that that is still here.

Comment (bsipocz) [reply astropy/utils/data.py]: ok, I'll put this back once travis passed.

**Comment by bsipocz** [comment]
@pllim - This is yours to merge. Previously passed on travis, I've only added back the two comments lines that was accidentally removed in a rebase conflict. https://travis-ci.org/astropy/astropy/builds/277040443

**Pull request merged**